pkg/driver/node/credentialprovider/provider_pod.go

```diff
@@ -141,7 +160,53 @@ func (c *Provider) findPodServiceAccountRole(ctx context.Context, provideCtx Pro
 
 // podLevelServiceAccountTokenName returns service account token name for Pod-level identity.
 // It escapes from slashes to make this token name path-safe.
-func podLevelServiceAccountTokenName(podID string, volumeID string) string {
+func podLevelServiceAccountTokenName(podID string, volumeID string) string { // TODO: Consider reusability with podLevelEksPodIdentityServiceAccountTokenName or at least renaming this one.
 	id := escapedVolumeIdentifier(podID, volumeID)
 	return id + ".token"
 }
+
+// podLevelEksPodIdentityServiceAccountTokenName returns service account token name for Pod-level identity with EKS Pod Identity.
+// It escapes from slashes to make this token name path-safe.
+func podLevelEksPodIdentityServiceAccountTokenName(podID string, volumeID string) string {
+	id := escapedVolumeIdentifier(podID, volumeID)
+	return id + "-eks-pod-identity.token"
+}
+
+// createIRSACredentialsEnvironment creates an environment with the environment variables needed for pod-level authentication with IRSA
+func (c *Provider) createIRSACredentialsEnvironment(ctx context.Context, provideCtx ProvideContext) (envprovider.Environment, error) {
+	roleARN, err := c.findPodServiceAccountRole(ctx, provideCtx)
```

Comment:
I think here there is a small issue related to my Pod Sharing change where currently `findPodServiceAccountRole()` will make API call to get ServiceAccount if `ServiceAccountEKSRoleARN` from MountpointS3PodAttachment custom resource is empty string. 

https://github.com/awslabs/mountpoint-s3-csi-driver/blob/main/pkg/driver/node/credentialprovider/provider_pod.go#L116-L119



We should not make this API call and instead always return `provideCtx.ServiceAccountEKSRoleARN` if we are using Pod Mounter. And if it's empty string we should return `Missing role annotation` error.

(In Pod Mounter (v2), we've decided to get role ARN from this custom resource instead of making Get `ServiceAccount` API call on Node)